### PR TITLE
don't hardcode changelog type to be markdown when sending to cf

### LINKED ISSUE #40: changelog processing is different from curseforge packager (for manual changelog), on curseforge
CF packager:
https://www.curseforge.com/wow/addons/dynamicboxer/files/2784081

changelog is as the original txt file
( https://github.com/mooreatv/DynamicBoxer/blob/master/ChangeLog.txt )

Bigwigs:
https://www.curseforge.com/wow/addons/dynamicboxer/files/2791311

lines are collapsed into paragraph
(and isn't doing markdown either)

is there an option to not process the manual changelog or to process it the same way as cf's packager?

### PR DESCRIPTION
**Pull request by mooreatv**
fixes #40

### CODE REVIEW

File: release.sh [hunk 1]
```diff
@@ -572,7 +572,7 @@ fi
 package=
 manual_changelog=
 changelog=
-changelog_markup=plain
+changelog_markup=text
```
Comment (mooreatv) [inline release.sh]: I checked it's safe to make that change as changelog_markup is only used below (beside tested for being equal to "markdown" or not)

Comment (nebularg) [reply release.sh]: quote that while you're there
```suggestion
changelog_markup="text"
```

File: release.sh [hunk 2]
```diff
@@ -1672,7 +1672,7 @@ if [ -z "$changelog" ]; then
 	changelog_markup="markdown"
 fi
 if [ -n "$manual_changelog" ] && [ -f "$topdir/$changelog" ]; then
-	echo "Using manual changelog at $changelog"
+	echo "Using manual changelog at $changelog (type $changelog_markup)"
```
Comment (nebularg) [inline release.sh]:
```suggestion
	echo "Using manual changelog at $changelog ($changelog_markup)"
```

**Comment by nebularg** [comment]
Thanks, I just looked at the output on the site and assumed it would have converted your list if it was sent as markdown

File: release.sh (hunk 1 above)
Comment (mooreatv) [reply release.sh]: was keeping it consistent with what it was (and some other variables like  repository_type) but sure

**Comment by mooreatv** [comment]
> Thanks, I just looked at the output on the site and assumed it would have converted your list if it was sent as markdown

yw. not following. I prefer text than mangled/not working markdown (even if github converts it much better, minus extra lines)

> Also check the value parsed from the pkgmeta and save "text" instead of "plain" (https://github.com/BigWigsMods/packager/blob/master/release.sh#L707)

The way it was working until this patch was it was either "markdown" or something else, and for something else it would still do markdown for cf

So not sure what you expect to change (beside maybe change your doc if you have a doc that says "plain" somewhere) ?

**Comment by nebularg** [comment]
Upload API is text, html or markdown; pkgmeta is plain (default), html, or markdown

**Comment by mooreatv** [comment]
if you are refering to
  https://authors.curseforge.com/knowledge-base/projects/527-preparing-the-packagemeta-file
I am guessing that
  https://authors.curseforge.com/knowledge-base/projects/529-api
is more accurate/recent

and I guess if cf ever fix their build system back we'll know, I made a test ^

**Comment by mooreatv** [comment]
also who is going to specify text instead of using the extension or the default? do you know of any pkgmeta doing so?

**Comment by nebularg** [comment]
I'll just do it, jesus

**Comment by mooreatv** [comment]
it's a real joy to help your project

**Comment by Nevcairiel** [comment]
The docs clearly specify a difference between pkgmeta and upload API, no amount of arguing is going to make that go away, so all you do is waste everyones time with it.

**Pull request closed without merging**